src/CPUSolver.cpp

```diff
@@ -748,6 +748,11 @@ void CPUSolver::tallyScalarFlux(segment* curr_segment, int azim_index,
   FP_PRECISION* sigma_t = curr_segment->_material->getSigmaT();
   FP_PRECISION delta_psi, exponential;
 
+  /* Extract weights for all polar angles */
+  FP_PRECISION weights[_num_polar];
```

Comment:
So we are using allocating `weights` on the stack each and every time `CPUSolver::tallyScalarFlux(...)` is called? Might it not be better to call `_quad->getWeight(...)` directly within the nested loop over energies and groups? If you inline this `Quadrature::getWeight()` I doubt that this would lead to a performance hit, and would reduce the code in this method by a few lines.